core-kotlin-modules/core-kotlin-collections-list/src/test/kotlin/com/baeldung/transformation/CustomObjectTransformationKtTest.kt

```diff
@@ -0,0 +1,188 @@
+package com.baeldung.transformation
+
+import org.junit.Test
+
+import org.junit.jupiter.api.Assertions.*
+import kotlin.random.Random
+
+class CustomObjectTransformationKtTest {
+
+    private val sourceEmpNameList = listOf(
+        Pair("John", "Doe"),
+        Pair("Jill", "Fowler"),
+        Pair("Jack", "Smith"),
+        Pair("John", "Doe"),
+    )
+
+    private val empList = sourceEmpNameList.mapIndexed { index, names ->
+        Employee(index, names.first, names.second, Random.nextDouble(1000.0, 5000.0))
+    }
+
+    private val empByOrgList = listOf(
+        Organization(
+            100,
+            listOf(
+                Employee(1, "John", "Doe", 1200.0),
+                Employee(2, "Jill", "Fowler", 1200.0)
+            )
+        ),
+        Organization(
+            200,
+            listOf(
+                Employee(3, "Jack", "Smith", 1200.0),
+                Employee(4, "John", "Doe", 1200.0)
+            )
+        ),
+    )
+
+    private val expectedEmpNames = sourceEmpNameList.map { "${it.first} ${it.second}" }
+
+    @Test
+    fun givenListOfEmployeeObjects_whenTransformedUsingMap_returnListOfEmployeeName() {
+        val empNameList = empList.map { it.name() }
+        assertNotNull(empNameList)
```

Comment:
As Eric has said we should avoid this extra check.

Reply:
changed per comments